crates/ruff/src/linter.rs

```diff
@@ -50,26 +51,29 @@ impl<T> LinterResult<T> {
     }
 }
 
+pub type DiagnosticsAndImports = (Vec<Diagnostic>, FxHashMap<PathBuf, Vec<Import>>);
+pub type MessagesAndImports = (Vec<Message>, FxHashMap<PathBuf, Vec<Import>>);
```

Comment:
These type seem to be somewhat common too. It could make sense to wrap them in a newtype-wrapper to avoid exposing internals (and it gives you the possibility to implement additional methods on them)

Reply:
Up to you. My tendency is to at least wrap types that are public.